ads/inabox/inabox-host.md

```diff
@@ -14,4 +14,103 @@ See the License for the specific language governing permissions and
+One goal of AMPHTML ads is for advertisers to create ad once and use 
+everywhere. This guide provides details about how an ad network can 
+join us to serve and render AMPHTML ads in various environments, for 
+example regular web & native mobile apps.
+
+## Terminologies
+
+###### AMPHTML ad
+Ad creatives that are written following the
+ [AMPHTML ads spec](https://amp.dev/documentation/guides-and-tutorials/learn/a4a_spec).
+ 
+###### inabox
+A term describes the situation that an AMPHTML ad is being served
+on a non-AMP environment, e.g a regular web page inside an iframe embed 
+or a native mobile app inside a WebView. This is very different from how
+ AMPHTML ad gets rendered on an AMP page, where the page and ad shares 
+ the same JS runtime.
+
+###### host
+The web page or native mobile app that the ad is being served on.
+
+###### friendly iframe
+An iframe with the same origin as the host page, so that it has
+full access to the host page content.
+
+###### cross-domain iframe
+An iframe on the host page that has different origin, so it can only
+communicate to the host page via postMessage API.
+
+## JS SDK (a.k.a ads tag) integration
+This is the case when AMPHTML ad is being served and rendered on a
+regular web page (non-AMP) inside an iframe. The iframe can be either
+a friendly iframe, or a cross-domain iframe. 
+
+##### Friendly iframe
+Using friendly iframe requires zero to little investment to get it
+ work, but it can be more involved if you want to get the full benefit
+ from AMPHTML ads, which will be discussed in the optimization session.
+  Because AMPHTML is trustful & secure, it can be put inside
+ a friendly iframe without worrying it hijacking the page
+and ruining the end user experience. Meantime, friendly iframe rendering
+in general requires less browser resources than cross-domain iframe rendering,
+which not only brings performance benefit but also revenue gain. One 
+thing to keep in mind though is that the ad network should have their own 
+mechanism to prevent pub
+modifying the ad content or faking clicks. Google Ad Manager uses this
+ approach to integrate with AMPHTML ads.
+
+Main work flow:
+1. The ads tag makes an ad request.
+1. The response contains a flag indicating it is an AMPHTML ad.
+1. The ads tag creates an iframe, and set the response content 
+to the iframe using `srcdoc`.
+
+Note that this friendly iframe approach right now is in an experimental
+stage. Put the following meta tag in the head of the ad HTML to opt-in 
+the experiment:
+`<meta name="amp-experiments-opt-in" content="inabox-viewport-friendly">`.
+
+###### Security assurance
+In case that the AMPHTML ad is generated by a 3rd party, to ensure it's
+secure to put it in a friendly iframe, the following 2 enforcements 
+should be considered:
+1. Add [Content-Security-Policy](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP)
+ rules to the ad response to 1) only load JS files from trusted AMP 
+ CDN; 2) disallow iframe embed.
```

Comment:
If you look on a page with GPT you can see Google is currently using:

```
<meta http-equiv="Content-Security-Policy" content="script-src https://cdn.ampproject.org/;object-src 'none';child-src blob:;frame-src 'none'">
```

Why not recommend that here?  If you're not careful, for example, you can break amp-bind by not allowing workers to be loaded from blobs.

See https://github.com/ampproject/amphtml/issues/21681 where I proposed AMP recommend this CSP.